common/smoke-test/Initialize-SmokeTests.ps1

```diff
@@ -90,20 +90,28 @@ function Set-EnvironmentVariables {
 
 function New-DeployManifest {
   Write-Verbose "Detecting samples..."
-  $javascriptSamples = (Get-ChildItem -Path "$repoRoot/sdk/$ServiceDirectory/*/samples/javascript/" -Directory
-    | Where-Object { Test-Path "$_/package.json" })
+  $packageDir = Get-ChildItem -Directory -Path "$repoRoot/sdk/$ServiceDirectory/*"
+
+  $javascriptSamples = $packageDir | ForEach-Object {
+    $versions = (Get-Item "$_/samples/*").Name
+    $newestVer = $versions | Sort-Object {[int]($_ -replace '^v' -replace '(\d+).*', '$1')} -Descending | Select-Object -First 1
+    return Get-ChildItem -Path "$_/samples/$newestVer/javascript/" -Recurse -Include package.json
+  }
 
   $manifest = $javascriptSamples | ForEach-Object {
-    # Example: azure-sdk-for-js/sdk/appconfiguration/app-configuration/samples/javascript
+    # Example: azure-sdk-for-js/sdk/appconfiguration/app-configuration/samples/v1/javascript
     @{
       # Package name for example "app-configuration"
-      Name               = ((Join-Path $_ ../../) | Get-Item).Name;
+      Name               = ((Join-Path $_ ../../../../) | Get-Item).Name;
```

Comment:
I'd suggest it is about time we refactor this path into a variable with a name and share it so we don't have this magic relative path in multiple places.